### PR TITLE
Windows additions

### PR DESCRIPTION
**Pull request by mikkeldamsgaard**
Support for UDP/TCP/UART on windows.
Support for Windows 64 Arms

### CODE REVIEW

File: src/resources/uart_win.cc [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+  OVERLAPPED _comm_events_overlapped{};
+  DWORD _event_mask = 0;
+
+  DWORD _error_code = ERROR_SUCCESS;
+};
+
+class UARTResourceGroup : public ResourceGroup {
+public:
+  TAG(UARTResourceGroup);
+  explicit UARTResourceGroup(Process* process, EventSource *event_source)
```
Comment (erikcorry) [inline src/resources/uart_win.cc:179]:
```suggestion
  explicit UARTResourceGroup(Process* process, EventSource* event_source)
```

**Review by erikcorry — APPROVED** [review]

**Comment by erikcorry** [comment]
Please merge up to get more build bot greenness.

File: src/resources/tcp_win.cc [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+  ARGS(ByteArray, proxy, TCPSocketResource, tcp_resource);
+  USE(proxy);
+
+  if (tcp_resource->closed()) return process->program()->null_object();
+
+  if (!tcp_resource->ready_for_read()) return Smi::from(-1);
+
+  if (!tcp_resource->receive_read_response()) WINDOWS_ERROR;
+
+  if (tcp_resource->read_count() == 0) return process->program()->null_object();
```
[the remark this answers is not in the document]
Comment (mikkeldamsgaard) [reply src/resources/tcp_win.cc:437]: No. Actually a read_count of 0 always means that the socket has been closed normally.
I will add a comment.

**Comment by mikkeldamsgaard** [comment]
Changed underscode positions for private fields.

**Pull request merged**